app/models/unit.rb

```diff
@@ -111,6 +111,18 @@ def cache_ancestry
                              end
   end
 
+  def semester_type
+    return unless program?
+
+    return 'yearly' if /(T[ı,i]p|D[ı,i][s,ş] Hekim|Hukuk)/i.match?(name)
```

Comment:
` /(T[ıi]p|D[ıi][s,ş] Hekim|Hukuk)/i` (Düzenli ifadelerde aralık için `-` kullanılmazsa düz liste olur, bu haliyle `,` karakteri de eşleme yapar)